cli/internal/cmd/upgradecheck.go

```diff
@@ -685,16 +684,16 @@ type versionFetcher interface {
 }
 
 // newCLIVersions returns a list of versions of the CLI which are a valid upgrade.
-func (v *versionCollector) newCLIVersions(ctx context.Context) ([]string, error) {
-	cliVersion, err := conSemver.New(constants.VersionInfo())
+func (v *versionCollector) newCLIVersions(ctx context.Context) ([]consemver.Semver, error) {
+	cliVersion, err := consemver.New(constants.VersionInfo())
 	if err != nil {
 		return nil, fmt.Errorf("parsing current CLI version: %w", err)
 	}
 	list := versionsapi.List{
 		Ref:         v.flags.ref,
 		Stream:      v.flags.stream,
 		Granularity: versionsapi.GranularityMajor,
-		Base:        fmt.Sprintf("v%d", cliVersion.Major),
+		Base:        fmt.Sprintf("v%d", cliVersion.Major()),
 		Kind:        versionsapi.VersionKindCLI,
 	}
 	minorList, err := v.versionsapi.FetchVersionList(ctx, list)
```

Comment:
Can we rework this method to return a slice of `Semver` instead?
I think this could simplify a lot of the error handling made here.

Reply:
From private discussions: I agree that this would be a nice change. However, this would require quite some changes in the versionsapi pkg to make the change consistent. Only changing the type of List.Versions is not enough. This could be done in a separate PR if we decide that this is a useful improvement.